### PR TITLE
Added Hoc

### PR DESCRIPTION
**Pull request by shanushka**
added HOC for fetching data through API

### CODE REVIEW

File: src/components/JobStory.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        <Link to='/comments'>
+          <span className='comments'>
+            comments ({story.commentIdArray ? story.commentIdArray.length : 0})
+          </span>
+        </Link>
+      </div>
+    </li>
+  ));
+
+    return (
```
Comment (sanjeevkpandit) [inline src/components/JobStory.js:25]: indentation?

File: src/hoc/WithFetchData.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+      this.state = {
+        stories: []
+      };
+    }
+
+    componentDidMount() {
+      this.getDataFromApi();
+
+      this.mounted = true;
```
Comment (sanjeevkpandit) [inline src/hoc/WithFetchData.js]: you can use a `loading` variable in the state.

File: src/hoc/WithFetchData.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    componentWillUnmount() {
+      this.mounted = false;
+    }
+
+    getDataFromApi = () => {
+      axios
+        .get(`https://hacker-news.firebaseio.com/v0/${story}.json?print=pretty`)
+        .then(res => {
+          res.data.forEach(result => {
```
Comment (sanjeevkpandit) [inline src/hoc/WithFetchData.js]: what if there is an error while fetching data?

File: src/hoc/WithFetchData.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            axios
+              .get(
+                `https://hacker-news.firebaseio.com/v0/item/${result}.json?print=pretty`
+              )
+              .then(response => {
```
Comment (sanjeevkpandit) [inline src/hoc/WithFetchData.js]: same here

File: src/hoc/WithFetchData.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+      this.mounted = true;
+    }
+    componentWillUnmount() {
+      this.mounted = false;
+    }
+
+    getDataFromApi = () => {
+      axios
+        .get(`https://hacker-news.firebaseio.com/v0/${story}.json?print=pretty`)
```
Comment (sanjeevkpandit) [inline src/hoc/WithFetchData.js]: Use constants to store the links

File: src/utils/getTimeDifference.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  const msPerMinute = 60 * 1000;
+  const msPerHour = msPerMinute * 60;
+  const msPerDay = msPerHour * 24;
+  const msPerMonth = msPerDay * 30;
+  const msPerYear = msPerDay * 365;
+
+  const currentTimestamp = Date.now();
+  const elapsed = currentTimestamp - previousTimestamp * 1000;
+
+  if (elapsed < msPerMinute) {
```
Comment (sanjeevkpandit) [inline src/utils/getTimeDifference.js:11]: `switch` may be better here

File: src/components/Comments.js [hunk 7]
```diff
@@ -1,19 +1,27 @@
...
-    super();
-    this.state = {};
-  }
-
   render() {
     return (
       <div>
-      <Tab></Tab>
+        <Tab />
+        <div>
```
Comment (sanjeevkpandit) [inline src/components/Comments.js]: extra unnecessary div?

File: src/components/Comments.js [hunk 8]
```diff
@@ -1,19 +1,27 @@
...
+          <div className='comments-overlay clearfix'>
+            <div className='title'>{this.props.location.state.story.title}</div>
```
Comment (sanjeevkpandit) [inline src/components/Comments.js:12]: Use header tags instead.

File: src/components/SingleComment.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+      time: ''
+    };
+  }
+
+  componentDidMount() {
+    this.getComment();
+  }
+
+  getComment() {
+    getCommentsByApi(this.props.id).then(data => {
```
Comment (sanjeevkpandit) [inline src/components/SingleComment.js:27]: error hander?

File: src/components/SingleComment.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        time: getTimeDifference(data.time)
+      });
+    });
+  }
+
+  render() {
+    return (
+      <div className='singlecomment' key={this.state.id} id={this.state.id}>
+        <li>
+          <div className='user'>
```
Comment (sanjeevkpandit) [inline src/components/SingleComment.js:44]: div inside li?

File: src/components/SingleComment.js [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        commentIdArray: data.kids,
+        time: getTimeDifference(data.time)
+      });
+    });
+  }
+
+  render() {
+    return (
+      <div className='singlecomment' key={this.state.id} id={this.state.id}>
+        <li>
```
Comment (sanjeevkpandit) [inline src/components/SingleComment.js]: li without ul?

File: src/utils/getTimeDifference.js (hunk 6 above)
Comment (shanushka) [reply src/utils/getTimeDifference.js:11]: swtch tyo comparision ma mildian rahecha

**Review by sanjeevkpandit — APPROVED** [review]

**Pull request merged**